allennlp/predictors/predictor.py

```diff
@@ -57,17 +63,157 @@ def predict_json(self, inputs: JsonDict) -> JsonDict:
         instance = self._json_to_instance(inputs)
         return self.predict_instance(instance)
 
+    def interpret_from_json(self, inputs: JsonDict) -> JsonDict:
+        """
+        Uses the gradients from :func:`get_gradients` to provide 
+        normalized interpretations for specific models. 
+
+        Raises
+        ------
+        NotImplementedError
+            This method should be overriden by the specific predictor class of each
+            predictor.
+        """
+        raise NotImplementedError
+
+    def attack_from_json(self, inputs: JsonDict) -> JsonDict:
+        """
+        Uses the gradients from :func:`get_gradients` to provide 
+        adversarial attacks for specific models. 
+
+        Raises
+        ------
+        NotImplementedError
+            This method should be overriden by the specific predictor class of each
+            predictor.
+        """
+        raise NotImplementedError
+
+    def get_model_predictions(self, inputs: JsonDict) -> List[Instance]:
+        """
+        Converts incoming json to a :class:`~allennlp.data.instance.Instance`
+        and adds labels to the :class:`~allennlp.data.instance.Instance`s given
```

Comment:
This would be clearer if you explicitly said that you run the model as the second step, before "and adds labels".